transaction/transaction.go

```diff
@@ -0,0 +1,65 @@
+package transaction
+
+import (
+	"database/sql"
+)
+
+// db is the global database connection object.
+var db *sql.DB
```

Comment:
I would like to avoid use of global varaiables where possible. Instead of having this global variable I'd suggest to have a new `Manager` type, something like this:

```go
// ManagerBuilder contains the data and logic needed to create
// a transaction manager.
type ManagerBuilder struct {
        logger logging.Logger
        db *sql.DB
        ...
}

// Manager knows how to start, resolve and in general manage database
// transactions.
type Manager struct {
        logger logging.Logger
        db *sql.DB
        ...
}

// NewManager creates a builder that can then be used to configure and
// create a new transaction manager.
func NewManager() *Manager {
        return &ManagerBuilder{}
}

// Logger sets the object that the manager will use to write messages to the log.
func (b *ManagerBuilder) Logger(value logging.Logger) *ManagerBuilder {
        b.logger = value
        return b
}

// Handle sets the database handle that the manager will use to create transactions.
func (b *ManagerBuilder) Handle(value *sql.DB) *ManagerBuilder {
        b.handle = value
        return b
}

// Build uses the data stored in the builder to create a new transaction manager.
func (b *ManagerBuilder) Build(ctx context.Context) (result *Manager, err error) {
        ...
}
```